### DISCUSSION TITLE
20240222 Weekly Development Meeting

### DISCUSSION THREAD

**Discussion by josecelano** [nautilus-cyberneering/.github#83] [Meetings]
- Time:  9:30 in Canary Islands (WET (UTC+0)).
- Time: 10:30 in Vaduz (CET (UTC+1)).

Location:
<https://meeting.blockfinance-eco.li/NautilusCyberneeringWeeklyDevMeeting>

Expected duration: 2 to 4 hours with breaks

Communicated attendants:

- CGarnham(@da2ce7)  - 
- JCelano(@josecelano) - x
- CBosse(@cgbosse) - x
- MMelchor(@mario-nt) - x
- GByrne(@grmbyrn) - x
- MVanDijke(@WarmBeer) - x

-------------------------------------------------------------

## Agenda:

1. Everyone will take their turn and:
- Go over his notes added as comments for the meeting which will have been added in advance to give an update of all the work done during the previous week indicating.
	- The issues worked on
	- The progress made on them
	- Other things done
- Inform of any faced challenges and request help if necessary to work on specific issues.
- Highlight the encountered difficulties, especially if there is a risk of them causing delays or affecting end product's quality.
- Communicate what will be the next week's tasks.
- List any new topics for further discussion in an additional Technical meeting that he noted for future discussion in comments to this discussion either before the meeting or after his turn so that they get considered.

During the meeting every team member will take their turns and follow this sequence.
The topics for further discussion will be noted in advance by each the team member and to the meetings agenda.

2. At the end of the meeting if there any new technical topics to be discussed further it will be decided to
set another meeting date for the discussion of these or to hold the meeting hereafter.

Also the persons to be part of the meeting will be decided on so that the others will be free to continue their own work.

**Comment by josecelano** [comment]
### Torrust Tracker

Overhaul logs using `tracing` crate format.

- [x] https://github.com/torrust/torrust-tracker/issues/700
- [x] https://github.com/torrust/torrust-tracker/issues/696
- [X] https://github.com/torrust/torrust-tracker/issues/697
- [X] https://github.com/torrust/torrust-tracker/issues/698

Examples:

```
2024-02-19T11:30:21.061293933+00:00 [HEALTH CHECK API][INFO] request; method=GET uri=/health_check request_id=78933bbf-c4cf-4897-b972-4c0fd252159a
2024-02-19T11:30:21.070329733+00:00 [HEALTH CHECK API][INFO] response; latency=9 status=200 OK request_id=78933bbf-c4cf-4897-b972-4c0fd252159a

2024-02-19T13:29:02.301023716+00:00 [HTTP TRACKER][INFO] request; server_socket_addr=0.0.0.0:7070 method=GET uri=/scrape?info_hash=%44%3C%76%02%B4%FD%E8%3D%11%54%D6%D9%DA%48%80%84%18%B1%81%B6 request_id=2c4aa57d-dd12-4cb8-95d1-e8193627c106
2024-02-19T13:29:02.301095545+00:00 [HTTP TRACKER][INFO] response; server_socket_addr=0.0.0.0:7070 latency=0 status=200 OK request_id=2c4aa57d-dd12-4cb8-95d1-e8193627c106
```

All logs include:

- Request ID (a UUID) in req and response (link response to request).
- Server socket to identify the service processing the request.
- UPPERCASE service target. For example: `HEALTH CHECK API`.

Discussions:

- https://github.com/torrust/torrust-tracker/issues/705

### Torrust Index

- [X] https://github.com/torrust/torrust-index/issues/455

### Torrust Index GUI

Focus on UI small issues:

- [x] https://github.com/torrust/torrust-index-gui/issues/456
- [x] https://github.com/torrust/torrust-index-gui/issues/452
- [x] https://github.com/torrust/torrust-index-gui/issues/451
- [x] https://github.com/torrust/torrust-index-gui/issues/449
- [x] https://github.com/torrust/torrust-index-gui/issues/409
- [x] https://github.com/torrust/torrust-index-gui/issues/352

Issues opened by users:

- [ ] https://github.com/torrust/torrust-index-gui/issues/483 **WIP**

Update dependencies:

- [x] https://github.com/torrust/torrust-index-gui/pull/472
- [x] https://github.com/torrust/torrust-index-gui/issues/393

**Comment by mario-nt** [comment]
### GUI
  - https://github.com/torrust/torrust-index-gui/issues/401
  

> - Created components for each part of the license
> - Getting the text from the license file and display it on the GUI - WIP
> - Tried out different ways of displaying the license data


### Index
	
- https://github.com/torrust/torrust-index/issues/445
>- https://github.com/torrust/torrust-index/pull/486
- https://github.com/torrust/torrust-index/issues/445

### Training/Research

- Vue composables
- Fetch API
- Getting text from file 
- Tidy first? book

**Comment by grmbyrn** [comment]
## Torrust Website
* Solutions to make Table of Contents highlight which section of the blog was highlighted.
* Found [Svelte-ToC](https://toc.janosh.dev/), read up on the docs, and how [Matt Fantinel](https://fantinel.dev/) used it.
* Implemented it on the Torrust website.
* Currently working on bug which prevents home button in header from working.
* Found [another](https://github.com/swyxio/swyxkit/blob/main/src/routes/%5Bslug%5D/TableOfContents.svelte) option if Svelte-ToC bug can't be resolved.

## Research
* Revising React through [projects](https://github.com/grmbyrn/github-api-finder).
